src/legacy/core_plugins/telemetry/server/get_telemetry_opt_in.ts

```diff
@@ -43,4 +57,50 @@ export async function getTelemetryOptIn(request: any) {
 
     throw error;
   }
+
+  const { attributes }: { attributes: SavedObjectAttributes } = savedObject;
+
+  // if enabled is already null, return null
+  if (attributes.enabled == null) return null;
```

Comment:
It's against my nature, since `==` catches `undefined` as well as `null`.  I could be convinced otherwise, but don't remember doing a `=== null` in a long time, unless I was specifically wanting to check `null` vs `undefined` for some reason.

Reply:
In what valid situation would `attributes.enabled` be `undefined`, and why should we return `null` in that situation?

Reply:
If someone (maybe us, accidently??) reset the saved object attributes to `{}`, the `enabled` field would be `undefined`, not null.  In questionable cases like this (as well as if the semver wasn't parseable), I've made some arbitrary decisions, and in most cases, you'd want this to be considered "re-prompt the user". 

It's easy for me to imagine folks wanting to defeat banner-ism by hacking saved objects, so ... I've tried to be pretty conservative here in our checks of things users might be "providing" to use :-)

Reply:
> If someone (maybe us, accidently??) reset the saved object attributes to {}, the enabled field would be undefined, not null.

In this scenario the fact that `enabled` is undefined is a bug and IMO we shouldn't act like it means something it doesn't. Looseness like this leads to subtle bugs that are hard to fix.

That said, the stakes are low in this context, so if you don't want to make the change we can simply agree to disagree.

Reply:
I think we'll have to agree to disagree at this point, but would welcome some kind of project-wide guidance on this.  As noted elsewhere, I think I lived for a while with a linter than MADE you use `== null`, so I'll admit my brain may be warped.

Still, seems like unless you are specifically expecting a `null` OR `undefined`, as semantically different things, from an API, you should always just use `== null` for such checks.  Also note, there are plenty of folks who shorten this check to `!x` (from `x == null`), since it's often the case that this works as well, but those ones do kinda bug me :-)